src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -393,9 +540,86 @@ private static DataGroup convertStarTableToDataGroup(StarTable table,
+    }*/
+
+    record EvalVal(long blank, boolean scaled, boolean hasBlank, double scale, double zero) {
+        public Number evalValue(Number val) {
+            if (hasBlank && val.doubleValue() == blank) return null;
+            if (!scaled) return val;
+            return val.doubleValue() * scale + zero;
+        }
     }
 
-    private static void addRowToDataGroup(StarTable table, DataGroup dataGroup, LinkedHashMap<ColumnInfo, Integer> colIdxMap, long rowIdx, String strategy) {
+    //This function is based on the packageValue function from the FitsStarTable class in the uk.ac.starlink.fits package
+    private static Object getValAsObject(Object elem, int icol, boolean[] hasBlank, long[] blanks, boolean[] isScaled,
+                                         double[] scales, double[] zeros) throws FitsException {
+        if (elem == null) {
+            return null;
+        } else {
+            Class bcls = elem.getClass();
+            Class cls = bcls.isArray() ? bcls.getComponentType() : null;
+            if (cls != null && Array.getLength(elem) == 1) {
+                EvalVal evaluator = new EvalVal(blanks[icol], isScaled[icol], hasBlank[icol], scales[icol], zeros[icol]);
+                switch (cls.toString()) {
+                    case "byte" -> {
+                        return evaluator.evalValue(((byte[])(elem))[0]);
+                    }
+                    case "short" -> {
+                        return evaluator.evalValue(((short[])(elem))[0]);
+                    }
+                    case "int" -> {
+                        return evaluator.evalValue(((int[])(elem))[0]);
+                    }
+                    case "long" -> {
+                        return evaluator.evalValue(((long[])(elem))[0]);
+                    }
+                    case "float" -> {
+                        return evaluator.evalValue(((float[])(elem))[0]);
+                    }
+                    case "double" -> {
+                        return evaluator.evalValue(((double[])(elem))[0]);
+                    }
+                    case "boolean" -> {
+                        return Boolean.valueOf(((boolean[])(elem))[0]);
+                    }
+                    case "class java.lang.String" -> {
+                        String val = ((String[])(elem))[0];
+                        return isEmpty(val) ? null : val;
+                    }
+                    default -> {
+                        logger.error("Unrecognized class type in FITS table file entry");
+                        throw new FitsException(
+                                "Unrecognized class type in FITS table file entry: " + cls.toString());
+                    }
+                }
+            }
+
+            if (cls != null && cls.isArray()) {
+                return ArrayFuncs.flatten(elem);
+            } else {
+                return elem instanceof String && isEmpty((String)elem) ? null : elem;
+            }
+        }
+    }
```

Comment:
`getValAsObj` is not clear. After a *lot* study. I realized it is about 4 four cases of `elem`
- `elem` is null
- `elem` is not an array
- `elem` is an array of 1
- `elem` is an array > 1

Here is my propose re-write:
```java
    private static Object getValAsObject(Object elem, int icol, boolean[] hasBlank, long[] blanks, boolean[] isScaled,
                                         double[] scales, double[] zeros) throws FitsException {
        if (elem == null) {
            return null;
        }
        else if (!elem.getClass().isArray()) {
            return elem instanceof String && isEmpty((String)elem) ? null : elem;
        }
        else if (Array.getLength(elem) == 1) {
            String cls = elem.getClass().getComponentType().toString();
            EvalVal evaluator = new EvalVal(blanks[icol], isScaled[icol], hasBlank[icol], scales[icol], zeros[icol]);
            return switch (cls) {
                case "byte" -> evaluator.evalValue(((byte[])(elem))[0]);
                case "short" -> evaluator.evalValue(((short[])(elem))[0]);
                case "int" -> evaluator.evalValue(((int[])(elem))[0]);
                case "long" -> evaluator.evalValue(((long[])(elem))[0]);
                case "float" -> evaluator.evalValue(((float[])(elem))[0]);
                case "double" -> evaluator.evalValue(((double[])(elem))[0]);
                case "boolean" -> Boolean.valueOf(((boolean[])(elem))[0]);
                case "class java.lang.String" -> isEmpty(((String[])(elem))[0]) ? null : ((String[])(elem))[0];
                default -> throw new FitsException( "Unrecognized class type in FITS table file entry: " + cls);
            };
        }
        else {
            return ArrayFuncs.flatten(elem);
        }
    }
```